Start adding sitemap documentation (#766)
sharetribe/flex-docs: src/docs/concepts-development/sitemap/index.md

---
title: Sitemap support in Sharetribe
slug: sitemap-in-sharetribe
updated: 2023-10-12
category: concepts-development
ingress:
  This guide describes how the Sharetribe tooling supports creating a
  sitemap for your marketplace.
published: true
---

> 

Reviewer: Useful links for people who read this article:

- https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap#xml
- https://developers.google.com/search/docs/crawling-indexing/sitemaps/large-sitemaps
- https://developers.google.com/search/docs/crawling-indexing/robots/intro

> Having a sitemap on your marketplace helps search engines to process

Reviewer 2: ```suggestion
Having a sitemap on your marketplace helps search engines crawl and index
```

your website correctly. For a traditional website, sitemaps are
relatively easy to make, because all of your routes are built in to the
application.

> 

Reviewer 2: Comment:
Maybe we could mention that we provide API endpoints (and link to them) for generating dynamic sitemaps already in the introduction section right now they're only mentioned under "Separate sitemaps for static and dynamic content"

However, on a marketplace, it is likely that you will have listing URLs
that include dynamic data from each listing – and you still want search
engines to index those pages.

<info>

The Sharetribe Web Template has tooling to create a dynamic sitemap by
default. Read more about the template sitemap structure and logic:

> **[Sitemap in Sharetribe Web Template](/sitemap-in-template)**

Reviewer: ```suggestion
**[Sitemap in Sharetribe Web Template](/ftw/sitemap-in-template/)**
```
Linkki ei toiminut.

</info>

If you are creating your own custom application, you can consider
whether you want to use the same practices that the web template uses to
craft the sitemap.

## Separate sitemaps for static and dynamic content

> The web template exposes three different sub-sitemaps from

Reviewer 2: Comment:
How this is worded makes it seem that this information is web-template specific. Is the idea here to use the approach the web-template uses as an example for non-template developers? What do you think about using a more general approach, e.g.:

Static content sitemaps help search engines quickly index unchanging pages like your "About" page, while dynamic content sitemaps focus on frequently updated or user-generated content, such as listings in your marketplace. For example, the Sharetribe Web Template generates three different sub-sitemaps linked from _sitemap-index.xml_, all with slightly different data:

_sitemap-index.xml_, all with slightly different data:

- _sitemap-default.xml_ exposes public built-in pages in the template.
  Non-public routes that require authentication are disallowed in the
  _robots.txt_ file.
- _sitemap-recent-listings.xml_ exposes routes to the most recently
  created listings. The listing data is fetched from a specific
  [_sitemapData_ endpoint](TODO API REF LINK). The endpoint returns the

>   10 000 most recent listings.

Reviewer 2: I couldn't find the answer via the API ref, but maybe we could have an answer for the question "what if I have more than 10 000 listings in my marketplace" here

Author: Great question, asked [in Slack](https://sharetribe.slack.com/archives/C03MHA74QQ4/p1697118379278729)

Author: No pagination in the sitemap endpoints, so folks who want to include listings beyond 10k will need to fetch the older listings through Integration API. I'll include a note on this in the article!
